packages/protocol/contracts/libs/LibDeploy.sol

```diff
@@ -37,4 +39,34 @@ library LibDeploy {
             OwnableUpgradeable(proxy).transferOwnership(owner);
         }
     }
+
+    function deployERC1967Proxy(
+        address impl,
+        address owner,
+        bytes memory data,
+        address timelock
+    )
+        internal
+        returns (address proxy)
+    {
+        proxy = deployERC1967Proxy(impl, owner, data);
+        if (
+            timelock != address(0) && owner != OwnableUpgradeable(proxy).owner()
```

Comment:
her we should require timelock != address(0) and timelock != OwnableUpgradeable(proxy).owner()

Reply:
Changed